ast/compile_test.go

```diff
@@ -3674,6 +3679,11 @@ func TestCompilerBuildComprehensionIndexKeySet(t *testing.T) {
 					]
 				}
 			`,
+			// Note: no comprehension index for line 7 (`z = [ ...`)
+			expected: exp{9: {
```

Comment:
This expectation is new: the nested comprehension is now indexed because `y`, previously hidden in the surrounding body of the `z` comprehension, is no now taken into account.

Reply:
This looks correct to me. We can index the inner comprehension but not the outer one. There was an issue in the original indexing implementation that resulted in this test case/fix: https://github.com/open-policy-agent/opa/issues/2497